packages/web-components/src/dialog/dialog.ts

```diff
@@ -79,76 +32,15 @@ export class Dialog extends FASTElement {
    * @public
    * The type of the dialog modal
    */
-  @attr({ attribute: 'modal-type' })
-  public modalType: DialogModalType = DialogModalType.modal;
-
-  /**
-   * @public
-   * Indicates whether the dialog is open
-   */
-  @attr({ mode: 'boolean' })
-  public open: boolean = false;
-
-  /**
-   * @public
-   * Indicates whether the dialog has a title action
-   */
-  @attr({ mode: 'boolean', attribute: 'no-title-action' })
-  public noTitleAction: boolean = false;
-
-  /**
-   * @private
-   * Indicates whether focus should be trapped within the dialog
-   */
-  private trapFocus: boolean = false;
-
-  /**
-   * @public
-   * Method called when the 'open' attribute changes
-   */
-  public openChanged(oldValue: boolean, newValue: boolean): void {
-    if (newValue !== oldValue) {
-      if (newValue && !oldValue) {
-        this.show();
-      } else if (!newValue && oldValue) {
-        this.hide();
-      }
-    }
-  }
-
-  /**
-   * @public
-   * Method called when the 'modalType' attribute changes
-   */
-  public modalTypeChanged(oldValue: DialogModalType, newValue: DialogModalType): void {
-    if (newValue !== oldValue) {
-      if (newValue == DialogModalType.alert || newValue == DialogModalType.modal) {
-        this.trapFocus = true;
-      } else {
-        this.trapFocus = false;
-      }
-    }
-  }
-
-  /**
-   * @public
-   * Method to set the component's state based on its attributes
-   */
-  public setComponent(): void {
-    if (this.modalType == DialogModalType.modal || this.modalType == DialogModalType.alert) {
-      this.trapFocus = true;
-    } else {
-      this.trapFocus = false;
-    }
-  }
+  @attr
+  public type: DialogType = DialogType.modal;
 
   /**
    * @public
    * Method to emit an event when the dialog's open state changes
-   * @param dismissed - Indicates whether the dialog was dismissed
    */
-  public onOpenChangeEvent = (dismissed: boolean = false): void => {
-    this.$emit('onOpenChange', { open: this.dialog.open, dismissed: dismissed });
+  public emitOpenChange = (): void => {
+    this.$emit('open', { open: this.dialog.open });
```

Comment:
Is this the type of event we want? https://developer.mozilla.org/en-US/docs/Web/API/EventSource/open_event https://html.spec.whatwg.org/multipage/server-sent-events.html#eventsource seems to indicate that a connection has been made to a service rather than a dialog being open/closed.

It would appear that the only native event for `dialog` is https://developer.mozilla.org/en-US/docs/Web/API/HTMLDialogElement/close_event

Reply:
Yeah, this would be a custom event with the same name - I think this is a good note, I'm curious given this is scoped to dialog if we think there's concerns around the API. It's directly related to the open state of the dialog which would allow folks to take action based on that change. I think the biggest concern here may be bubbling and conflicting with a listener for a service...though I'm not sure how significant that might be. Alternatives for pithy names that describe this well? I'm not a huge fan of prefixing with the element simply because you'd have to add the listener on the element, which makes it kind of repetitive.

Reply:
I'm flexible here. I thinkt he liklihood of collision is low, but we could align more with HTML `[popover]` and add a `toggle` event. That event offers `event.oldState` and `event.newState` on the event object, which would be a change for us but nothing too difficult. Popover also emits a `beforetoggle` event, which is something @radium-v suggested which could be something we add later if necessary.

Reply:
Presumably the reason that we only have a native "close" event is because the app/website is controlling when the dialog is opened, therefore external to the dialog we already have this information. However, if a close button is inside the dialog and part of encapsulation like we're doing here, we need to surface that event through the `<dialog>` element. I would opt for keeping this as close to standards as we can unless there is a reason to change it, and only `$emit` a "close" event.

Having said that I have no strong opinion on which way we go, toggle event seems like it should work fine and get us both open and closed state updates. Perhaps we should do both? Otherwise we may lose the `<dialog>` "close" event since it does not bubble.

Reply:
I've been mulling this around and I think I'm going to update this to `toggle` and then we can all see how that feels. I think it has a couple advantages:

- `toggle` wouldn't conflict with `EventSource.onopen` 
- `toggle` is a better name than `open` because it covers both actions
- `toggle` might be a good abstraction to cover over the non-uniform surface of `HTMLDialogElement` events (the non-existent "open" event and the `close` event)

I think @janechu makes a great point about feature parity and event bubbling but my current thinking right now is that consumers might prefer a uniform event. I'll push this work soon and we can mull it over. If it still feels like we should only have a `close` event, then I'm happen to roll back to that.

Reply:
Logging a private convo here for transparency and future genreation's sake:

I asked OpenUI if there was any discussion about backporting HTML `[popover]`'s `toggle` and `beforetoggle` events to `HTMLDialogElement` and lo and behold there has been discussion!

- Discussed at TPAC in Sept 2023 https://github.com/whatwg/html/issues/9733
- A PR for HTML https://github.com/whatwg/html/pull/10091
- A Chromium implementation https://chromium-review.googlesource.com/c/chromium/src/+/5238172
- A Webkit patch https://github.com/WebKit/WebKit/pull/23332
- A Firefox bug filed https://bugzilla.mozilla.org/show_bug.cgi?id=1876762

It seems like this is skating to where the puck is going to be. I've updated the event system to be more inline with the update by supplying `{old|new}State = 'open'|'closed'` to the `event.detail` object.